### PR TITLE
Address deprecation warnings in Storybook components

### PR DESCRIPTION
**Pull request by robinmetral**
## Purpose

☝️ 

## Approach and changes

- address deprecation warnings (`noMargin`) in Storybook components
- migrate a couple components to TypeScript
- remove an unused SB component

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: docs/features/icons.stories.mdx [hunk 1]
```diff
@@ -4,10 +4,8 @@ import { Meta, Intro, Icons } from '../../.storybook/components';
 
 # Icons
 
-<Intro as="div">
-
-The icons used throughout Circuit UI come from SumUp's icon library, [`@sumup/icons`](Packages/icons), which is a required peer dependency of `@sumup/circuit-ui`. Refer to its [documentation](Packages/icons) for usage instructions.
```
Comment (connor-baer) [inline docs/features/icons.stories.mdx:9]: 👀 Why this change?

**Review by connor-baer — APPROVED** [review]

File: docs/features/icons.stories.mdx (hunk 1 above)
Comment (robinmetral) [reply docs/features/icons.stories.mdx:9]: Ah I was trying to fix the Intro component after I refactored it (because it was hardcoding cascading typography values), but forgot to add this doc back. Thanks! Will link to the docs, though probably out of the `Intro` to avoid having to do the same thing as before

Comment (robinmetral) [reply docs/features/icons.stories.mdx:9]: Close enough without hacky CSS, will move forward (happy to think of alternatives later if we want to, right now I'd favor simplicity)

Screenshot 2022-01-28 at 22 00 08
